src/Microsoft.DotNet.Cli.Utils/CommandResolution/PackagedCommandSpecFactory.cs

```diff
@@ -68,17 +89,22 @@ internal PackagedCommandSpecFactory(Action<string, IList<string>> addAdditionalA
                 commandArguments,
                 depsFilePath,
                 commandResolutionStrategy,
-                nugetPackagesRoot,
+                packageFolders,
                 runtimeConfigPath);
         }
 
         private string GetCommandFilePath(
-            string nugetPackagesRoot,
+            IEnumerable<string> packageFolders,
             LockFileTargetLibrary toolLibrary,
             LockFileItem runtimeAssembly)
         {
-            var packageDirectory = new VersionFolderPathResolver(nugetPackagesRoot)
-                .GetInstallPath(toolLibrary.Name, toolLibrary.Version);
+
+            var packageFoldersCount = packageFolders.Count();
+            var userPackageFolder = packageFoldersCount == 1 ? string.Empty : packageFolders.First();
+            var fallbackPackageFolders = packageFoldersCount > 1 ? packageFolders.Skip(1) : packageFolders;
+
+            var packageDirectory = new FallbackPackagePathResolver(userPackageFolder, fallbackPackageFolders)
+                .GetPackageDirectory(toolLibrary.Name, toolLibrary.Version);
```

Comment:
If the tool's nupkg folder was deleted ``packageDirectory`` will be null. I would expect this to fail with a message to restore again. ``GetPackageDirectory`` returns null for any package that wasn't found.

Currently it looks like Path.Combine would throw with path1 cannot be null.